torchci/pages/nightlies.tsx

```diff
@@ -0,0 +1,453 @@
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import dayjs from "dayjs";
+import ReactECharts from "echarts-for-react";
+import { EChartsOption } from "echarts";
+import {
+  GridRenderCellParams,
+  GridValueFormatterParams,
+} from "@mui/x-data-grid";
+import useSWR from "swr";
+import {
+  Grid,
+  Paper,
+  TextField,
+  Typography,
+  Stack,
+  Skeleton,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  SelectChangeEvent,
+} from "@mui/material";
+import { LocalizationProvider, DateTimePicker } from "@mui/x-date-pickers";
+
+import { useEffect, useState } from "react";
+
+import { RocksetParam } from "lib/rockset";
+import { fetcher } from "lib/GeneralUtils";
+
+import TablePanel from "components/metrics/panels/TablePanel";
+import { durationDisplay } from "components/TimeUtils";
+
+function NightlyJobsRedPanel({ params, repo }: { params: RocksetParam[], repo: string }) {
+
+  let repo_p = params.find(({ name }) => name == "repo");
+  if(repo_p && repo) repo_p.value = repo; 
+
+  const url = `/api/query/nightlies/nightly_jobs_red?parameters=${encodeURIComponent(
+    JSON.stringify(params)
+  )}`;
+
+  const { data } = useSWR(url, fetcher, {
+    refreshInterval: 5 * 60 * 1000, // refresh every 5 minutes
+  });
+
+  if (data === undefined) {
+    return <Skeleton variant={"rectangular"} height={"100%"} />;
+  }
+
+  const options: EChartsOption = {
+    title: { text: "% "+repo+" nightly jobs by red" },
+    grid: { top: 48, right: 8, bottom: 24, left: 36 },
+    dataset: { source: data },
+    xAxis: { type: "time" },
+    yAxis: {
+      type: "value",
+      axisLabel: {
+        formatter: (value: number) => {
+          return (value * 100).toString() + "%";
+        },
+      },
+    },
+    series: [
+      {
+        type: "bar",
+        encode: {
+          x: "granularity_bucket",
+          y: "red",
+        },
+      },
+    ],
+    tooltip: {
+      trigger: "axis",
+      valueFormatter: (value: any) => {
+        return (value * 100).toFixed(2) + "%";
+      },
+    },
+  };
+
+  return (
+    <Paper sx={{ p: 2, height: "100%" }} elevation={3}>
+      <ReactECharts
+        style={{ height: "100%", width: "100%" }}
+        option={options}
+      />
+    </Paper>
+  );
+}
+
+function MasterCommitRedPanel({ params }: { params: RocksetParam[] }) {
```

Comment:
Did I miss anything as I couldn't find this function used anywhere? Probably a copy/paste thing from `metrics.tsx`?